### PR TITLE
Update system test docs to include new Elastic Agents

### PR DESCRIPTION
**Pull request by mrodm**
Part of #787 

This PR updates the documentation related to System tests to include information about the independent Elastic Agents developed, still in technical preview.

### CODE REVIEW

File: docs/howto/system_testing.md [hunk 1]
```diff
@@ -411,6 +428,7 @@ for system tests.
 | service_notify_signal | string |  | Signal name to send to 'service' when the test policy has been applied to the Agent. This can be used to trigger the service after the Agent is ready to receive data. |
 | skip.link | URL |  | URL linking to an issue about why the test is skipped. |
 | skip.reason | string |  | Reason to skip the test. If specified the test will not execute. |
+| skip_ignored_fields | array string |  | List of fields to be skipped when perfoing mapping validation. |
```
Comment (mrodm) [inline docs/howto/system_testing.md:431]: Adding field introduced in #1738

File: docs/howto/system_testing.md [hunk 2]
```diff
@@ -40,7 +41,9 @@ Packages have a specific folder structure (only relevant parts shown).
 
 To define a system test we must define configuration on at least one level: a package or a data stream's one.
 
-First, we must define the configuration for deploying a package's integration service. We can define it on either the package level:
+If the package does not require a service for testing, there is no need to define any service deployer.
```
Comment (mrodm) [inline docs/howto/system_testing.md:44]: Note about not requiring anymore a service deployer.

**Review by jsoriano — APPROVED** [review]

File: docs/howto/system_testing.md [hunk 3]
```diff
@@ -402,6 +410,15 @@ for system tests.
 
 | Option | Type | Required | Description |
 |---|---|---|---|
+| agent.linux_capabilities | array string | | Linux Capabilities that must been enabled in the system to run the Elastic Agent process. |
```
Comment (jsoriano) [inline docs/howto/system_testing.md:413]:
```suggestion
| agent.linux_capabilities | array string | | Linux Capabilities that must be enabled in the system to run the Elastic Agent process. |
```

File: docs/howto/system_testing.md [hunk 4]
```diff
@@ -402,6 +410,15 @@ for system tests.
...
+| agent.pid_mode | string | | Turns on sharing between container and the host operating system the PID address space. |
+| agent.ports | array string | | List of ports to be exposed to access to the Elastic Agent.|
+| agent.runtime | string | | Runtime to run Elastic Agent process. |
+| agent.pre_start_script.language | string | | Programming language of the provisioning script. Default: `bash`.|
+| agent.pre_start_script.contents | string | | Code to run as a provisioning script. |
```
Comment (jsoriano) [inline docs/howto/system_testing.md:418]:
```suggestion
| agent.pre_start_script.contents | string | | Code to run before starting the agent. |
```

File: docs/howto/system_testing.md [hunk 5]
```diff
@@ -402,6 +410,15 @@ for system tests.
...
+| agent.provisioning_script.language | string | | Programming language of the provisioning script. Currently, just supported `sh`. |
+| agent.provisioning_script.contents | string | | Code to run as a provisioning script. |
```
Comment (jsoriano) [inline docs/howto/system_testing.md:420]:
```suggestion
| agent.provisioning_script.contents | string | | Code to run as a provisioning script to customize the system where the agent will be run. |
```

File: docs/howto/system_testing.md [hunk 6]
```diff
@@ -402,6 +410,15 @@ for system tests.
 
 | Option | Type | Required | Description |
 |---|---|---|---|
+| agent.linux_capabilities | array string | | Linux Capabilities that must been enabled in the system to run the Elastic Agent process. |
+| agent.pid_mode | string | | Turns on sharing between container and the host operating system the PID address space. |
+| agent.ports | array string | | List of ports to be exposed to access to the Elastic Agent.|
+| agent.runtime | string | | Runtime to run Elastic Agent process. |
+| agent.pre_start_script.language | string | | Programming language of the provisioning script. Default: `bash`.|
```
Comment (jsoriano) [inline docs/howto/system_testing.md:417]:
```suggestion
| agent.pre_start_script.language | string | | Programming language of the pre-start script, executed before starting the agent. Default: `bash`.|
```

Comment (jsoriano) [inline docs/howto/system_testing.md:417]: Maybe we should also default it to `sh`, given that it is what is used in the provisioning script?

File: docs/howto/system_testing.md [hunk 7]
```diff
@@ -402,6 +410,15 @@ for system tests.
...
+| agent.pre_start_script.contents | string | | Code to run as a provisioning script. |
+| agent.provisioning_script.language | string | | Programming language of the provisioning script. Currently, just supported `sh`. |
+| agent.provisioning_script.contents | string | | Code to run as a provisioning script. |
+| agent.user | string | | User that runs the Elastic Agent proces. |
```
Comment (jsoriano) [inline docs/howto/system_testing.md:421]:
```suggestion
| agent.user | string | | User that runs the Elastic Agent process. |
```

File: docs/howto/system_testing.md (hunk 1 above)
Comment (jsoriano) [reply docs/howto/system_testing.md:431]: Thanks!

Comment (jsoriano) [inline docs/howto/system_testing.md:431]:
```suggestion
| skip_ignored_fields | array string |  | List of fields to be skipped when performing validation of fields ignored during ingestion. |
```

File: docs/howto/system_testing.md [hunk 8]
```diff
@@ -40,7 +41,9 @@ Packages have a specific folder structure (only relevant parts shown).
 
 To define a system test we must define configuration on at least one level: a package or a data stream's one.
 
-First, we must define the configuration for deploying a package's integration service. We can define it on either the package level:
+If the package does not require a service for testing, there is no need to define any service deployer.
+If the package requires a service, we must define the configuration for deploying a package's integration service.
+We can define it on either the package level:
```
[the remark this answers is not in the document]
Comment (mrodm) [reply docs/howto/system_testing.md:46]: I prefer your suggestion, I'll update it, thanks!

File: docs/howto/system_testing.md (hunk 6 above)
Comment (mrodm) [reply docs/howto/system_testing.md:417]: I mixed the default languages for both scripts.
In the pre-start script, it must be `sh`. This is the only supported for now.
For the provisioning script, there is no restriction and the default one is bash.

I'll update those default values.

Comment (jsoriano) [reply docs/howto/system_testing.md]: Could we make both to be `sh` by default? It looks weird to have different defaults for them.

File: docs/howto/system_testing.md [hunk 9]
```diff
@@ -402,6 +409,15 @@ for system tests.
 
 | Option | Type | Required | Description |
 |---|---|---|---|
+| agent.linux_capabilities | array string | | Linux Capabilities that must be enabled in the system to run the Elastic Agent process. |
+| agent.pid_mode | string | | Turns on sharing between container and the host operating system the PID address space. |
```
Comment (jsoriano) [inline docs/howto/system_testing.md:413]: Maybe we should mention the possible values here. And phrase it in a way that is less bound to containers.
```suggestion
| agent.pid_mode | string | | Controls access to PID namespaces. When set to `host`, the agent will have access to the PID namespace of the host. |
```

**Review by jsoriano — APPROVED** [review]

File: docs/howto/system_testing.md (hunk 6 above)
Comment (mrodm) [reply docs/howto/system_testing.md]: Yes, it could be done like that, no problem.
I'll update elastic-package and package-spec :+1:

Comment (mrodm) [reply docs/howto/system_testing.md]: Indeed, in code it was set for both scripts `sh` as default language:
- https://github.com/elastic/elastic-package/blob/7b156a51b8db9f300d4ba2fb67d6bcea9b744b42/internal/testrunner/runners/system/test_config.go#L113
- https://github.com/elastic/elastic-package/blob/7b156a51b8db9f300d4ba2fb67d6bcea9b744b42/internal/agentdeployer/info.go#L16

**Review by jsoriano — APPROVED** [review]

**Pull request merged**